ros2_socketcan/src/socket_can_receiver.cpp

```diff
@@ -171,18 +171,17 @@ CanId SocketCanReceiver::receive_fd(void * const data, const std::chrono::nanose
   // Read
   struct canfd_frame frame;
   const auto nbytes = read(m_file_descriptor, &frame, sizeof(frame));
+  const auto data_length = static_cast<CanId::LengthT>(frame.len);
+  const auto expected_length = sizeof(frame) - sizeof(frame.data) + data_length;
+
   // Checks
   if (nbytes < 0) {
     throw std::runtime_error{strerror(errno)};
   }
-  if (static_cast<std::size_t>(nbytes) < sizeof(frame)) {
+  if (static_cast<std::size_t>(nbytes) < expected_length) {
```

Comment:
Reading `frame.len` and `frame.data` before checking if the `read` succeeded is an unsafe memory access. Move the check to before these fields are accessed.

```suggestion
  if (nbytes < 0) {
    throw std::runtime_error{strerror(errno)};
  }

  const auto data_length = static_cast<CanId::LengthT>(frame.len);
  const auto expected_length = sizeof(frame) - sizeof(frame.data) + data_length;
  
  if (static_cast<std::size_t>(nbytes) < expected_length) {
```

Reply:
@knzo25, could you take a quick look at this?  :pray:

Reply:
Ooh, crap, though there was an implicit initialization in the struct definition :sweat_smile:

Reply:
(can not accept the suggestion though from the web page. Will push it from here)

Reply:
Addressed in [73bae2f](https://github.com/autowarefoundation/ros2_socketcan/pull/50/commits/73bae2f667caedc938b38f89c97ffc2fbe8ba1c5)

Reply:
(remember to mark this as resolved)